File: week2/most-common-word.java
// Uses the multiset class in Multiset.java
class Solution {
    public String mostCommonWord(String paragraph, String[] banned) {
        // Add banned words to hashset so they can be accessed in constant time
        HashSet<String> bannedWords = new HashSet<>();
        for (String s : banned) {
            bannedWords.add(s);
        }
        // Add words in paragraph to multiset if not banned
        Multiset<String> paragraphWords = new Multiset<>();
        // Split paragraph by any non-alphanumeric occurring one or more times
        String[] words = paragraph.split("\\W+");
        for (String word : words) {
            word = word.replaceAll("[^A-Za-z]+", "").toLowerCase();
            if (!bannedWords.contains(word)) {
                paragraphWords.add(word);
            }
        }
        // Find word with highest count
        String mostCommonWord = "";
        int maxOccurrences = 0;
        for (String word : paragraphWords.elementSet()) {
            if (paragraphWords.count(word) > maxOccurrences) {
                mostCommonWord = word;
                maxOccurrences = paragraphWords.count(word);
            }
        }
        return mostCommonWord;
    }
}

File: 2026/week2/most-common-word.py
class Solution:
    def mostCommonWord(self, paragraph: str, banned: List[str]) -> str:
        paragraph = re.sub(r'[^\w\s]', ' ', paragraph.lower())
        words = Counter(paragraph.split())
        banned_set = set(banned)
        res = ''
        current_max = 0
        for key in words:
            if key not in banned_set and words[key] > current_max:
                current_max = words[key]
                res = key
        return res
